app/notes/[noteId]/page.tsx

```diff
@@ -9,15 +9,31 @@ type Props = {
 };
 export default async function NotePage({ params }: Props) {
   // Task: Restrict access to the note page only to the user who created the note
+
+  const noteId = Number((await params).noteId);
+
   // 1. Check if the sessionToken cookie exists
   const sessionTokenCookie = await getCookie('sessionToken');
 
-  // 2. Query the notes with the session token and noteId
+  // 2. Check if the note exists
+  const noteExists = await selectNoteExists(noteId);
+
+  if (!noteExists) {
+    return (
+      <div>
+        <h1>Error loading note {noteId}</h1>
+        <div>The note does not exist</div>
+        <Link href="/notes">Back to notes</Link>
+      </div>
+    );
+  }
+
+  // 3. Query the notes with the session token and noteId
   const note =
     sessionTokenCookie &&
     (await getNote(sessionTokenCookie, Number((await params).noteId)));
```

Comment:
this can be reused from `noteId` local variable

Reply:
d55def8e7a5cdfcf539d19fbad58b3eb6fe1e64f